### PR TITLE
adding github repo guidelines

### PR DESCRIPTION
**Pull request by adusumillipraveen**
image

AB#782

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: source/ways-of-working/new-component/github-repo.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+title: Create Github Repository
+last_reviewed_on: 2020-03-06
+review_in: 6 months
+weight: 3
+---
+
+# <%= current_page.data.title %>
+
+Any developer added to [hmcts org](https://github.com/hmcts) would be able to create a new Github repo.
```
Comment (timja) [inline source/ways-of-working/new-component/github-repo.html.md.erb]:
```suggestion
Any developer added to [hmcts org](https://github.com/hmcts) is able to create a new Github repo.
```

File: source/ways-of-working/new-component/github-repo.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+       <td>`<team-name>`</td>
+       <td>Write</td>
+     </tr>
+     <tr>
+       <td>`<team-name>-admins`</td>
+       <td>admin</td>
+     </tr>
+     <tr>
+       <td>[bots](https://github.com/orgs/hmcts/teams/bots) (super humans at HMCTS who have magical powers to do CI)</td>
+       <td>admin</td>
```
Comment (timja) [inline source/ways-of-working/new-component/github-repo.html.md.erb]: don't they just need write access?

File: source/ways-of-working/new-component/github-repo.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+     </tr>
+    </table>
+
+    `Note: You need to revoke the user level admin access you might have got while creating the repository.`
+    
+5. Add a branch protection rule for `master` branch with recommended settings
```
Comment (timja) [inline source/ways-of-working/new-component/github-repo.html.md.erb]: I would mention that you should never add an outside collaborator to a repository, they must join the org

File: source/ways-of-working/new-component/github-repo.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+# <%= current_page.data.title %>
+
+Any developer added to [hmcts org](https://github.com/hmcts) would be able to create a new Github repo.
+
+
+1. Naming convention for repository is `{product}-{component}`.
+2. Choose from a template in the drop down depending on whether it's a front end or backend component. These templates provide all the boiler plate code needed for a new app. 
+    - [spring-boot-template](https://github.com/hmcts/spring-boot-template)
+    - [expressjs-template](https://github.com/hmcts/expressjs-template)
+3. Repository should be public.
```
Comment (timja) [inline source/ways-of-working/new-component/github-repo.html.md.erb]: feels like we need a link to coding in the open? or maybe the gds guide to explain this?
i.e. you shouldn't create a private repo unless you really know what you're doing

File: source/ways-of-working/new-component/github-repo.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,43 @@
+---
+title: Create Github Repository
```
Comment (timja) [inline source/ways-of-working/new-component/github-repo.html.md.erb]: maybe?
```suggestion
title: Create a Github Repository
```

File: source/ways-of-working/new-component/github-repo.html.md.erb (hunk 2 above)
Comment (adusumillipraveen) [inline source/ways-of-working/new-component/github-repo.html.md.erb]:
```suggestion
       <td>write</td>
```

**Comment by timja** [comment]
You going to create a slack auto response for this page?

**Pull request merged**